lib/dropbox_api/connection_builder.rb

```diff
@@ -1,16 +1,17 @@
 module DropboxApi
   class ConnectionBuilder
-    def initialize(oauth_bearer)
+    def initialize(oauth_bearer, middleware: MiddleWare::Stack.new)
```

Comment:
Building up on my previous suggestion, the middleware wouldn't be passed in as a reference on initialization, but it would be initialized within the constructor and then exposed so `Client` can access it.

If I'm missing something and this isn't actually a good idea please let me know. You're the master on this PR.